File: Lecture-7/NumericalHollowPyramid.java
import java.util.Scanner;

public class NumericalHollowPyramid {
    public static void main(String[] args) {
        
        try (Scanner sc = new Scanner(System.in)) {

            System.out.print("Enter height: ");
            int height = sc.nextInt();

            for (int i = 1; i <= height; i++) {

                for (int k = 1; k <= height - i; k++) {
                    System.out.print(" ");
                }

                for (int j = 1; j <= (2 * i) - 1; j++) {

                    if ((i == 1 || i == height) || (j == 1 || j == (2 * i) - 1)) {
                        System.out.print(i);
                    } else {
                        System.out.print(" ");
                    }
                }

                System.out.println();
            }
        }
    }
}

File: DS-Algo-with-Python/Lecture-7/NumericalHollowPyramid.py
height = int(input("Enter height: "))

for i in range(1, height + 1):
    for k in range(1, height - i + 1):
        print(" ", end="")

    for j in range(1, 2 * i):
        if (i == 1 or i == height) or (j == 1 or j == (2 * i) - 1):
            print(i, end="")
        else:
            print(" ", end="")
    
    print("")
